src/webfaf/problems.py

```diff
@@ -382,44 +365,78 @@ def get_problems(filter_form):
-                          to_release=filter_form.to_release.data,
-                          probable_fix_osr_ids=probable_fix_osr_ids,
-                          bug_filter=filter_form.bug_filter.data,
-                          solution=filter_form.solution)
+    p = query_problems(db,
+                       hist_table,
+                       opsysrelease_ids,
+                       component_ids,
+                       associate_id,
+                       arch_ids,
+                       exclude_taintflag_ids,
+                       types,
+                       lambda query: (
+                           query.filter(hist_field >= since_date)
+                           .filter(hist_field <= to_date)),
+                       function_names=filter_form.function_names.data,
+                       binary_names=filter_form.binary_names.data,
+                       source_file_names=filter_form.source_file_names.data,
+                       since_version=filter_form.since_version.data,
+                       since_release=filter_form.since_release.data,
+                       to_version=filter_form.to_version.data,
+                       to_release=filter_form.to_release.data,
+                       probable_fix_osr_ids=probable_fix_osr_ids,
+                       bug_filter=filter_form.bug_filter.data,
+                       limit=pagination.limit,
+                       offset=pagination.offset,
+                       solution=filter_form.solution)
+    return p
+
+
+def problems_list_table_rows_cache(filter_form, pagination):
+    key = ",".join((filter_form.caching_key(),
+                    str(pagination.limit),
+                    str(pagination.offset)))
+
+    cached = flask_cache.get(key)
+    if cached is not None:
+        return cached
+
+    p = get_problems(filter_form, pagination)
+
+    cached = (render_template("problems/list_table_rows.html",
+                              problems=p), len(p))
+
+    flask_cache.set(key, cached, timeout=60*60)
+    return cached
 
 
 @problems.route("/")
 def dashboard():
+    pagination = Pagination(request)
+
     filter_form = ProblemFilterForm(request.args)
     if filter_form.validate():
-        p = list(get_problems(filter_form))
+        if request_wants_json():
+            p = get_problems(filter_form, pagination)
+        else:
+            list_table_rows, problem_count = \
+                problems_list_table_rows_cache(filter_form, pagination)
+
+            return render_template("problems/list.html",
+                                   list_table_rows=list_table_rows,
+                                   problem_count=problem_count,
+                                   filter_form=filter_form,
+                                   pagination=pagination)
     else:
         p = []
 
     if request_wants_json():
         return jsonify(dict(problems=p))
 
-    return Response(stream_with_context(
-        stream_template("problems/list.html",
```

Comment:
You can also remove the `stream_template` function from webfaf:
https://github.com/abrt/faf/blob/ba9c45e9ef61ed41cea02433b5cdd4343373019b/src/webfaf/utils.py#L373